rfc/halyard-lite.md

```diff
@@ -0,0 +1,475 @@
+# Replacement of Halyard
+
+| | |
+|-|-|
+| **Status**     | _**Proposed**, Accepted, Implemented, Obsolete_ |
+| **RFC #**      | [89](https://github.com/spinnaker/governance/pull/89) |
+| **Author(s)**  | Eric Zimanyi (@ezimanyi) |
+| **SIG / WG**   | Kubernetes SIG |
+
+## Overview
+
+The current recommended way to deploy Spinnaker is by using Halyard. While Halyard
+has greatly simplified some aspects of installing and configuring Spinnaker, it
+has a number of shortcomings that are particularly apparent when deploying to
+Kubernetes. This document proposes replacing Halyard with a smaller tool that
+integrates better into the Kubernetes ecosystem. Throughout this document, the new
+tool is referred to by the temporary name _Halyard Lite_.
+
+### Goals and Non-Goals
+
+Goals:
+* Improve the install process for users deploying to Kubernetes by providing an
+install path that is Kubernetes-native and integrates well with the Kubernetes
+ecosystem
```

Comment:
Interesting wording here. Today, Halyard is _not_ Kubernetes specific. It sounds like this proposal is saying that Kubernetes will be the only supported deployment path for Spinnaker through Halyard?

I understand there's Kubernetes-native, but have we figured out what the story will be for operators that use Halyard today, but not K8s?

Reply:
Following up on myself (and please correct if wrong): There's nothing in this proposal that makes Halyard necessarily Kubernetes-only, just that it's more heavily indexing on a smooth k8s experience.

Reply:
This is a good point, and I'll update the RFC to be a bit more clear about it.  In general the primary motivation here is to make the experience easier for k8s users, because the coupling of config generation and deployment makes it particularly difficult for them to use tools they are used to using.

But as the output of Halyard Lite will just be your config files and some metadata about the service versions you want to deploy, there's nothing here that ties it specifically to Kubernetes.  Users deploying to VMs can use the output as part of their deployment process.

Halyard will no longer handle actually deploying these services to VMs (which it just did via some bash wrapping an `apt-get` with a pinned version), but given that it could only deploy all services to the same VM as Halyard it's not clear to me that this was overly useful beyond test setups anyway.  Just as with k8s, it will be up to downstream tools/scripts to actually deploy the services using the config generated by Halyard.

Reply:
Added a section "Non-Kubernetes" to "Install Pathways"